app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJob.scala

```diff
@@ -62,37 +60,37 @@ class PollingNewMessageWithWorkItemJob @Inject()
 
   override def runJob(implicit ec: ExecutionContext): Future[RunningOfJobSuccessful] = {
     val now = dateTimeService.timestamp()
-    process(now.minus(1, ChronoUnit.DAYS), now, appConfig.maxRetryAttempts)
+    process(now.minus(appConfig.failureRetryAfter.toJava), now, appConfig.maxFailureRetryAttempts)
   }
 
   private def process(failedBefore: Instant, availableBefore: Instant, maximumRetries: Int): Future[RunningOfJobSuccessful] = {
-    workItemRepository.pullOutstanding(failedBefore, availableBefore)
+    workItemService.pullOutstanding(failedBefore, availableBefore)
       .flatMap {
-        case None => Future.successful(RunningOfJobSuccessful)
-        case Some(wi) =>
-          lazy val nextRunTime = wi.availableAt.plus(appConfig.runSubmissionWorkItemAfter.toJava)
-
-          getNewMessages(wi.item.exciseNumber).flatMap {
+        case None =>
+          Future.successful(RunningOfJobSuccessful)
 
-            case (true, NoMessageFound) if wi.failureCount < maximumRetries =>
-
-              workItemRepository.markAs(wi.id, ProcessingStatus.Failed, Some(nextRunTime))
+        case Some(wi) =>
 
-            case (true, NoMessageFound) =>
+          val ern = wi.item.exciseNumber
 
-              workItemRepository.completeAndDelete(wi.id)
+          getNewMessages(ern).flatMap {
 
-            case (true, MessageReceived) =>
-              workItemRepository.completeAndDelete(wi.id)
+            case MoreMessagesToGet =>
+              logger.info(s"[PollingNewMessageWithWorkItemJob] - Work item for ERN $ern has further outstanding messages and has been added back to the queue")
+              workItemService.markAs(wi.id, ProcessingStatus.ToDo, Some(wi.availableAt))
```

Comment:
is this line needed? It looks like here we are not doing anything so the workItem should already been in the Todo status and have the same availableAt

Reply:
It will have moved to InProgress status when the work item was picked up